stdlib/gc.mli

```diff
@@ -458,15 +458,45 @@ module Memprof :
        type of records passed to the callback triggered by the
        sampling of an allocation. *)
 
+    type ('minor, 'major) tracker = {
+      alloc_minor: allocation -> 'minor option;
+      alloc_major: allocation -> 'major option;
+      promote: 'minor -> 'major option;
+      dealloc_minor: 'minor -> unit;
+      dealloc_major: 'major -> unit;
+    }
+    (**
+       A [('minor, 'major) tracker] describes how memprof should track
+       sampled blocks over their lifetime, keeping a user-defined piece
+       of metadata for each of them: ['minor] is the type of metadata
+       to keep for minor blocks, and ['major] the type of metadata
+       for major blocks.
+
+       If an allocation-tracking or promotion-tracking function returns [None],
+       memprof stops tracking the corresponding value.
+     *)
+
+    val default_tracker:
+      ?alloc_minor:(allocation -> 'minor option) ->
+      ?alloc_major:(allocation -> 'major option) ->
+      ?promote:('minor -> 'major option) ->
+      ?dealloc_minor:('minor -> unit) ->
+      ?dealloc_major:('major -> unit) ->
+      unit -> ('minor, 'major) tracker
+    (** Default callbacks simply return [None] or [()] *)
+
+    val simple_tracker:
+      alloc:(allocation -> 'data option) ->
+      dealloc:('data -> unit) ->
+      unit ->
+      ('data, 'data) tracker
+    (** A simple tracker for just allocation and deallocation *)
```

Comment:
I added a trailing `unit` parameter to this signature, in case we might want to evolve the API by adding extra optional parameters later. (If you don't have a non-labelled parameter after your optional parameters, things are broken.)